localsdk/alarm/alarm.c

```diff
@@ -170,6 +170,11 @@ static int alarm_state_callback(LOCALSDK_ALARM_EVENT_INFO *eventInfo) {
 // Enable or disable alarm
 bool alarm_switch(bool state) {
     LOGGER(LOGGER_LEVEL_DEBUG, "Function is called...");
+    if (APP_CFG.alarm.enable == false) {
+        LOGGER(LOGGER_LEVEL_INFO, "Alarm switch ignored, because alarms are disabled");
+        return true;
```

Comment:
Exiting the function without the final debug message "Function completed...". This helps to look for a problem during memory leaks, do not ignore it. It is better to wrap the whole code in the condition:
```c
if(APP_CFG.alarm.enable == true) {
    ....
} else {
    LOGGER(LOGGER_LEVEL_INFO, "Alarm switch ignored, because alarms are disabled");
}